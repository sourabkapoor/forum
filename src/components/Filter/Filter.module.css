.filter {
  display: flex;
  justify-content: space-between;
  align-items: center;
}
.filter__categories {
  display: flex;
}
.filter__category {
  border-radius: 20px;
  /* background-color: #272729; */
  border: 1px solid transparent;
  margin-right: 8px;
  padding: 8px 10px;
  display: flex;
  align-items: center;
  column-gap: 4px;
  color: black;
}
.filter__category:hover {
  /* background-color: #343434; */
  cursor: pointer;
}
.filter__icon {
  width: 24px;
  height: 24px;
}

@media (max-width: 640px) {
  div.filter__category {
    display: none;
  }
}
