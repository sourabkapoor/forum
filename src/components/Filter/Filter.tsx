import CSSModules from "react-css-modules";
import Card from "../Card/Card";
import styles from "./Filter.module.css";
import { AiOutlineRocket } from "react-icons/ai";
import { HiOutlineFire } from "react-icons/hi";
import { MdOutlineNewReleases } from "react-icons/md";
import { MdOutlineLeaderboard } from "react-icons/md";
import { BsThreeDots } from "react-icons/bs";
import { BsFillInboxesFill } from "react-icons/bs";
import { IoIosArrowDown } from "react-icons/io";
import { DocumentData } from "firebase/firestore";
import { useState } from "react";

interface Props {
  filterRising: () => Promise<DocumentData>;
  filterNew: () => Promise<DocumentData>;
  filterTop: () => Promise<DocumentData>;
  addPosts: (promise: Promise<DocumentData>) => void;
}

const Filter: React.FC<Props> = ({
  filterRising,
  filterNew,
  filterTop,
  addPosts,
}) => {
  const [activeTab, setActiveTab] = useState(-1)

  const showPosts = (tab: number, func: Function) => {
    setActiveTab(tab)
    addPosts(func())
  }

  return (
    <Card>
      <div styleName="filter">
        <div styleName="filter__categories">
          <button
            styleName="filter__category"
            style={{background: activeTab === 0 ? "#d7bbf1" : "", color: activeTab === 0 ? "white" : ""}}
            onClick={() => showPosts(0, filterRising)}
          >
            <HiOutlineFire styleName="filter__icon" />
            <span>Rising</span>
          </button>
          <button styleName="filter__category" onClick={() => showPosts(1, filterNew) } 
            style={{background: activeTab === 1 ? "#d7bbf1" : "", color: activeTab === 1 ? "white" : ""}}
          >
            <MdOutlineNewReleases styleName="filter__icon" />
            <span>New</span>
          </button >
          <button styleName="filter__category" onClick={() => showPosts(2, filterTop)} 
            style={{background: activeTab === 2 ? "#d7bbf1" : "", color: activeTab === 2 ? "white" : ""}}
          >
            <MdOutlineLeaderboard styleName="filter__icon" />
            <span>Top</span>
          </button>
        </div>

        {/* <div styleName="filter__category">
          <BsFillInboxesFill styleName="filter__icon" />
          <IoIosArrowDown styleName="filter__icon" />
        </div> */}
      </div>
    </Card>
  );
};

export default CSSModules(Filter, styles, {
  allowMultiple: true,
  handleNotFoundStyleName: "log",
});
