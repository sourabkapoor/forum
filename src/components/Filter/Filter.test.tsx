import { queryByPlaceholderText, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import userEvent from "@testing-library/user-event";
import each from "jest-each";
import Filter from "./Filter";
import { useReducer } from "react";

describe("Filter", () => {
  test("snapshot", () => {
    const mockProps = {
      filterRising: jest.fn(),
      filterNew: jest.fn(),
      filterTop: jest.fn(),
      addPosts: jest.fn(),
    };
    const { asFragment } = render(<Filter {...mockProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
  test("filter buttons call all functions in props", async () => {
    const mockProps = {
      filterRising: jest.fn(),
      filterNew: jest.fn(),
      filterTop: jest.fn(),
      addPosts: jest.fn(),
    };
    render(<Filter {...mockProps} />);

    const filterRisingButton = screen.queryByRole("button", { name: "Rising" });
    const filterNewButton = screen.queryByRole("button", { name: "New" });
    const filterTopButton = screen.queryByRole("button", { name: "Top" });

    await userEvent.click(filterRisingButton as HTMLButtonElement);
    await userEvent.click(filterNewButton as HTMLButtonElement);
    await userEvent.click(filterTopButton as HTMLButtonElement);

    expect(mockProps.filterTop).toHaveBeenCalledTimes(1);
    expect(mockProps.filterNew).toHaveBeenCalledTimes(1);
    expect(mockProps.filterRising).toHaveBeenCalledTimes(1);
    expect(mockProps.addPosts).toHaveBeenCalledTimes(3);
  });
});
