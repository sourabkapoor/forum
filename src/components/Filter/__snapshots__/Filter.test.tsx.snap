// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="filter"
    >
      <div
        class="filter__categories"
      >
        <button
          class="filter__category"
        >
          <svg
            class="filter__icon"
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M17.657 18.657A8 8 0 016.343 7.343S7 9 9 10c0-2 .5-5 2.986-7C14 5 16.09 5.777 17.656 7.343A7.975 7.975 0 0120 13a7.975 7.975 0 01-2.343 5.657z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M9.879 16.121A3 3 0 1012.015 11L11 14H9c0 .768.293 1.536.879 2.121z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <span>
            Rising
          </span>
        </button>
        <button
          class="filter__category"
        >
          <svg
            class="filter__icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0V0z"
              fill="none"
            />
            <path
              d="M23 12l-2.44-2.78.34-3.68-3.61-.82-1.89-3.18L12 3 8.6 1.54 6.71 4.72l-3.61.81.34 3.68L1 12l2.44 2.78-.34 3.69 3.61.82 1.89 3.18L12 21l3.4 1.46 1.89-3.18 3.61-.82-.34-3.68L23 12zm-4.51 2.11l.26 2.79-2.74.62-1.43 2.41L12 18.82l-2.58 1.11-1.43-2.41-2.74-.62.26-2.8L3.66 12l1.85-2.12-.26-2.78 2.74-.61 1.43-2.41L12 5.18l2.58-1.11 1.43 2.41 2.74.62-.26 2.79L20.34 12l-1.85 2.11zM11 15h2v2h-2zm0-8h2v6h-2z"
            />
          </svg>
          <span>
            New
          </span>
        </button>
        <button
          class="filter__category"
        >
          <svg
            class="filter__icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M16 11V3H8v6H2v12h20V11h-6zm-6-6h4v14h-4V5zm-6 6h4v8H4v-8zm16 8h-4v-6h4v6z"
            />
          </svg>
          <span>
            Top
          </span>
        </button>
      </div>
      <div
        class="filter__category"
      >
        <svg
          class="filter__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4.98 1a.5.5 0 0 0-.39.188L1.54 5H6a.5.5 0 0 1 .5.5 1.5 1.5 0 0 0 3 0A.5.5 0 0 1 10 5h4.46l-3.05-3.812A.5.5 0 0 0 11.02 1H4.98zM3.81.563A1.5 1.5 0 0 1 4.98 0h6.04a1.5 1.5 0 0 1 1.17.563l3.7 4.625a.5.5 0 0 1 .106.374l-.39 3.124A1.5 1.5 0 0 1 14.117 10H1.883A1.5 1.5 0 0 1 .394 8.686l-.39-3.124a.5.5 0 0 1 .106-.374L3.81.563zM.125 11.17A.5.5 0 0 1 .5 11H6a.5.5 0 0 1 .5.5 1.5 1.5 0 0 0 3 0 .5.5 0 0 1 .5-.5h5.5a.5.5 0 0 1 .496.562l-.39 3.124A1.5 1.5 0 0 1 14.117 16H1.883a1.5 1.5 0 0 1-1.489-1.314l-.39-3.124a.5.5 0 0 1 .121-.393z"
          />
        </svg>
        <svg
          class="filter__icon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 294.1L383 167c9.4-9.4 24.6-9.4 33.9 0s9.3 24.6 0 34L273 345c-9.1 9.1-23.7 9.3-33.1.7L95 201.1c-4.7-4.7-7-10.9-7-17s2.3-12.3 7-17c9.4-9.4 24.6-9.4 33.9 0l127.1 127z"
          />
        </svg>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
